ext/reflection/php_reflection.c

```diff
@@ -1049,29 +1048,115 @@ static void reflection_attribute_factory(zval *object, zend_string *name, zval *
 	reflection_instantiate(reflection_attribute_ptr, object);
 	intern  = Z_REFLECTION_P(object);
 	reference = (attribute_reference*) emalloc(sizeof(attribute_reference));
-	reference->name = name;
-	reference->arguments = arguments;
+	reference->name = zend_string_copy(name);
+	ZVAL_COPY(&reference->arguments, arguments);
 	intern->ptr = reference;
 	intern->ref_type = REF_TYPE_ATTRIBUTE;
 }
 /* }}} */
 
+static int convert_ast_to_zval(zval *ret, zend_ast *ast, zend_class_entry *ce)
```

Comment:
Can you rename `ce` to `context_ce` here? I got confused why there is a class entry here since its used for all types of attributes.

Reply:
The most appropriate name for this would be "scope" I guess...